### DISCUSSION TITLE
Check IP already banned before banning across all jails

### DISCUSSION THREAD

**Discussion by exxosuk** [fail2ban/fail2ban#3849] [Ideas]
I know per jail F2B checks if a IP is already banned. Though can there be a more global check for IP banning from maybe IPtables directly ?

The "problem" I see, is when attackers do multiple attack vectors, they can instantly match several jails at once. It results in the same IP being blocked in multiple jails multiple times.

My jails are already rather large, so I don't want to condense them anymore. As if a IP gets banned by mistake, it becomes harder to troubleshoot what the problem jail/rule is.

**Comment by sebres** [comment]
Most jails uses the multi-port banning action by default, thus it is fully correct to do the "already banned" check per jail.

> It results in the same IP being blocked in multiple jails multiple times.

Yes, and mostly it'd affect different ports in that case. This is basically intended.

Vice versa, if the banning action is allports-action, then the IP must be banned in single jail only (first jail the IP gets found with `maxretry` attempts), because it is banned across all ports, so quasi affects all jails, and shall not connect anymore unless unbanned.

>  As if a IP gets banned by mistake, it becomes harder to troubleshoot what the problem jail/rule is.

How the considering of "Check IP already banned before banning across all jails" would help to troubleshoot what the problem is?

Additionally one can find what jail(s) the IP(s) got banned with this command:
```bash
fail2ban-client banned <IP> ... <IP>; # -- return list(s) of jails where given IP(s) are banned
```
